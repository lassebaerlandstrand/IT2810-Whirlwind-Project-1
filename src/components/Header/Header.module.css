.header {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 1rem;
  text-decoration: none;

  h1 {
    color: var(--primary-color);
    font-style: italic;
    margin: 0;
  }

  h2 {
    font-size: 0.75rem;
    color: var(--primary-color);
    font-style: italic;
    margin: -0.75rem 0 0 0;
  }
}

.img {
  width: 4rem;
  height: 4rem;
}
