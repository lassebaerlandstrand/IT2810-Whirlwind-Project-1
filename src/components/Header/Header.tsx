import { Link } from 'react-router-dom';
import styles from './Header.module.css';

const Header = () => {
  return (
    <header>
      <Link to="/" className={styles.header}>
        <img src="src\assets\Whirlwind Logo.png" alt="Whirlwind Logo" className={styles.img} />
        <div>
          <h1>Whirlwind</h1>
          <h6>Weather Forecast</h6>
        </div>
      </Link>
    </header>
  );
};

export default Header;
