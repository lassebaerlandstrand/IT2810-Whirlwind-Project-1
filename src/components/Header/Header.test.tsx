import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import Header from './Header';

describe('Header tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
