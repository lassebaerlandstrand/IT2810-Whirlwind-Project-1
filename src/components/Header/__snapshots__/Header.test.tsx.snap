// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header tests > should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <a
      class="_header_63c2ea"
      href="/"
    >
      <img
        alt="Whirlwind Logo"
        class="_img_63c2ea"
        src="/project1/src/assets/Whirlwind_Logo.png"
      />
      <div>
        <h1>
          Whirlwind
        </h1>
        <h2>
          Weather Forecast
        </h2>
      </div>
    </a>
  </header>
</DocumentFragment>
`;
